### PR TITLE
fix(oauth-providers): support latest hono

### PR DESCRIPTION
**Pull request by nakasyou**
In Hono before the update, the argument of c.set was also any, but after the update, I couldn't do it anymore, so I got a TypeScript error. So I fixed it.

### CODE REVIEW

File: .changeset/fluffy-lions-clap.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@hono/oauth-providers': major
```
Comment (yusukebe) [inline .changeset/fluffy-lions-clap.md:2]: This should be a `patch`.

Comment (neolectron) [reply .changeset/fluffy-lions-clap.md:2]: @nakasyou any updates on this PR ? If not available I would like to get back to it 👍 .

Comment (nakasyou) [reply .changeset/fluffy-lions-clap.md]: Sorry for very late response, I fixed it.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@nakasyou Thanks!

@neolectron Thanks for pinging.

**Pull request merged**